tensorboard/plugins/histogram/summary_test.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for the image plugin summary generation functions."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import hashlib
+import random
+
+import numpy as np
+import six
+import tensorflow as tf
+
+from tensorboard.plugins.histogram import metadata
+from tensorboard.plugins.histogram import summary
+
+
+class SummaryTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(SummaryTest, self).setUp()
+    tf.reset_default_graph()
+
+    np.random.seed(0)
+    self.gaussian = np.random.normal(size=[500])
+
+  def pb_via_op(self, summary_op, feed_dict=None):
+    actual_pbtxt = tf.Session().run(summary_op, feed_dict=feed_dict or {})
+    actual_proto = tf.Summary()
+    actual_proto.ParseFromString(actual_pbtxt)
+    return actual_proto
+
+  def get_summary_pb(self, name='nemo', data=None, bucket_count=None,
```

Comment:
The method name is misleading, it doesn't look like it's making eqality assertions. Normally when there is a method asserting behavior I expect "assert" to be in the name.

Maybe change to 

get_summary_pb_and_assert_pb_op_equality?

Very verbose I know :(